cylc/flow/task_job_mgr.py

```diff
@@ -645,25 +664,41 @@ def _run_job_cmd(self, cmd_key, suite, itasks, callback):
         Group itasks with their user@host.
         Put a job command for each user@host to the multiprocess pool.
 
+        n.b. When doing tasks like kill we may want to use the same host.
+        so we get the host and owner back from the itask and keep this working
+        using hosts rather than platforms.
         """
         if not itasks:
             return
+        # sort itasks into lists based upon where they were run.
         auth_itasks = {}
         for itask in itasks:
-            if (itask.task_host, itask.task_owner) not in auth_itasks:
-                auth_itasks[(itask.task_host, itask.task_owner)] = []
-            auth_itasks[(itask.task_host, itask.task_owner)].append(itask)
-        for (host, owner), itasks in sorted(auth_itasks.items()):
-            cmd = ["cylc", cmd_key]
+            # retrieve owner and host used last time
+            host = itask.summary['job_hosts'][max(itask.summary['job_hosts'])]
+            owner = ''
+            if 'owner' in itask.summary:
+                owner = itask.summary['owner']
+            platform_name = itask.platform['name']
+            if (platform_name, host, owner) not in auth_itasks:
+                auth_itasks[(platform_name, host, owner)] = []
+            auth_itasks[(platform_name, host, owner)].append(itask)
+
+        # Go through each list of itasks and carry out commands as required.
+        for (platform_n, host, owner), itasks in sorted(auth_itasks.items()):
+            platform = platform_from_name(platform_n)
+            if is_remote_host(platform['remote hosts'][0]):
```

Comment:
Use the swanky new `is_remote_platform` feature to centralise this (as we will likely change the implementation later).

Reply:
Done